modelon/impact/client/entities/case.py

```diff
@@ -127,6 +129,66 @@ def simulation_log_level(self, simulation_log_level: str):
         self._analysis['simulation_log_level'] = simulation_log_level
 
 
+class CustomArtifact:
+    """CustomArtifact class"""
+
+    def __init__(
+        self,
+        workspace_id: str,
+        experiment_id: str,
+        case_id: str,
+        data: Dict[str, str],
+        exp_sal,
+    ):
+        self._workspace_id = workspace_id
+        self._exp_id = experiment_id
+        self._case_id = case_id
+        self._data = data
+        self._exp_sal = exp_sal
+
+    @property
+    def id(self):
+        """Id of the custom artifact."""
+        return self._data['id']
+
+    @property
+    def download_as(self):
+        """File name for the downloaded artifact."""
+        return self._data['downloadAs']
+
+    def download(self, path: Optional[str] = None):
```

Comment:
I'd like to see, e.g., `get_artifact` or `get_artifact_data` returning stream as well without writing to file. This is to cover the case of small artifacts like json files that client can easily process in memory. For a multisim ~10000 explicitly downloading all artifacts will be big additional overhead.

Reply:
Ah ok. Would it be ok to have a method to_stream to do this?
so the workflow would look like
```
artifact = case.get_artifact(some_id)
stream=artifact.to_stream()
```

Reply:
I'd prefer `get_data` to indicate that it's not conversion but a request to server and it's not an asyncio stream but a full data object.